std-bits/table/src/main/java/org/enso/table/data/index/UnorderedMultiValueKey.java

```diff
@@ -49,6 +50,16 @@ protected Object getObjectFolded(int index) {
     return EnsoObjectWrapper.foldObject(this.get(index), textFoldingStrategy.get(index));
   }
 
+  /**
+   * Create an UnorderedMultiValueKey using a subset of the storages in this key.
+   * @param storageIndices a list of indices into the storages array describing
+   *                       which storages to keep.
+   * @return a new key with only the selected storages.
+   */
+  public UnorderedMultiValueKey subKey(int[] storageIndices) {
+    Storage<?>[] newStorages = Arrays.stream(storageIndices).mapToObj(i -> storages[i]).toArray(Storage<?>[]::new);
+    return new UnorderedMultiValueKey(newStorages, rowIndex, textFoldingStrategy);
+  }
```

Comment:
Note that this recomputes the hashes over the whole thing, right? This seems relatively expensive

Reply:
It is, and it probably makes up most of the work of computing an index in the first place, which is why this rewrite doesn't make a huge difference. We could consider something to generate multiple indices at once, using different column subsets, but that's a whole new level of complexity so it has to be worth it.